### PR TITLE
Blacklist Indicator with Trend Micro Apex from Microsoft Teams

### PR DESCRIPTION
**Pull request by jschipp-r7**
## Requirements

Workflow contributors, verify you have completed the following items by checking them off:

### Assessment

Post your results with screenshots in this PR using the following style:

```
$ icon-validate .
[*] Validating workflow at .

[*] Running Integration Validators...
[*] Executing validator WorkflowDirectoryNameMatchValidator
[*] Executing validator WorkflowFilesValidator
[*] Executing validator WorkflowHelpValidator
[*] Executing validator WorkflowChangelogValidator
[*] Executing validator WorkflowVendorValidator
[*] Executing validator WorkflowVersionValidator
[*] Executing validator WorkflowExtensionValidator
[*] Executing validator WorkflowSupportValidator
[*] Executing validator WorkflowPNGHashValidator
[*] Executing validator WorkflowICONFileNameValidator
[*] Executing validator WorkflowScreenshotValidator
[*] Executing validator WorkflowTitleValidator
[*] Executing validator WorkflowDescriptionValidator
[*] Executing validator WorkflowNameValidator
[*] Executing validator WorkflowProfanityValidator
[*] Executing validator WorkflowHelpPluginUtilizationValidator
[*] Executing validator WorkflowICONFileValidator
[*] Workflow successfully validated!

----
[*] Total time elapsed: 28.343ms
```

### Screenshots

Screenshots help your reviewers provide valuable feedback on your contribution. To add a screenshot here, simply drag and drop the PNG files from a file explorer.

api_config
teams_failure
teams1
teams2
teams3
trendmicro
workflow1

### CODE REVIEW

File: workflows/Blacklist_Indicators_with_Trend_Micro_Apex_from_Microsoft_Teams/Blacklist_Indicators_with_Trend_Micro_Apex_from_Microsoft_Teams.icon [hunk 1]
```diff
@@ -0,0 +1,2204 @@
...
+  "kom": {
+    "komandVersion": "1.54.0-0-gc18eb",
+    "komFileVersion": "2.0.0",
+    "exportedAt": "2020-06-22T23:39:38.221655678Z",
+    "workflowVersions": [
+      {
+        "name": "Blacklist Indicator with Trend Micro Apex from Microsoft Teams",
+        "type": "runnable",
+        "version": "",
+        "description": "This workflow will blacklist an IP or SHA1 indicator with Trend Micro Apex from a Microsoft Teams command.",
```
Comment (mrinehart-r7) [inline workflows/Blacklist_Indicators_with_Trend_Micro_Apex_from_Microsoft_Teams/Blacklist_Indicators_with_Trend_Micro_Apex_from_Microsoft_Teams.icon:11]:
```suggestion
        "description": "This workflow will blacklist an IP address or SHA-1 hash indicator with Trend Micro Apex from a Microsoft Teams command.",
```

File: workflows/Blacklist_Indicators_with_Trend_Micro_Apex_from_Microsoft_Teams/Blacklist_Indicators_with_Trend_Micro_Apex_from_Microsoft_Teams.icon [hunk 2]
```diff
@@ -0,0 +1,2204 @@
...
+          },
+          "19a25c50-657d-4a7e-bb3d-18a7460eaf16": {
+            "nodeId": "19a25c50-657d-4a7e-bb3d-18a7460eaf16",
+            "name": "Usage",
+            "type": "artifact",
+            "continueOnFailure": false,
+            "isDisabled": false,
+            "parameters": {
+              "input": {
+                "content": "This workflow will blacklist and unblacklist indicators in Trend Micro Apex. To use this workflow provide a blacklist or unblacklist command in Microsoft Teams with a supported indicator of IP address or SHA1 hash.\n<br><br>\nFor example:\n<br><br>\n<ul>\n<li>!blacklist-indicator a67f7c65c8ac1788a41be05f39d083ff532811a5</li>\n<li>!unblacklist-indicator a67f7c65c8ac1788a41be05f39d083ff532811a5</li>\n<li>!blacklist-indicator 198.51.100.10</li>\n<li>!unblacklist-indicator 198.51.100.10</li>\n</ul>"
```
Comment (mrinehart-r7) [inline workflows/Blacklist_Indicators_with_Trend_Micro_Apex_from_Microsoft_Teams/Blacklist_Indicators_with_Trend_Micro_Apex_from_Microsoft_Teams.icon:527]:
```suggestion
                "content": "This workflow will blacklist and unblacklist indicators in Trend Micro Apex. To use this workflow provide a blacklist or unblacklist command in Microsoft Teams with a supported indicator of IP address or SHA-1 hash.\n<br><br>\nFor example:\n<br><br>\n<ul>\n<li>!blacklist-indicator a67f7c65c8ac1788a41be05f39d083ff532811a5</li>\n<li>!unblacklist-indicator a67f7c65c8ac1788a41be05f39d083ff532811a5</li>\n<li>!blacklist-indicator 198.51.100.10</li>\n<li>!unblacklist-indicator 198.51.100.10</li>\n</ul>"
```

File: workflows/Blacklist_Indicators_with_Trend_Micro_Apex_from_Microsoft_Teams/Blacklist_Indicators_with_Trend_Micro_Apex_from_Microsoft_Teams.icon [hunk 3]
```diff
@@ -0,0 +1,2204 @@
...
+          },
+          "83d3ef05-d615-424f-89a1-4743468fb5bb": {
+            "nodeId": "83d3ef05-d615-424f-89a1-4743468fb5bb",
+            "name": "Unblacklist Failure",
+            "type": "artifact",
+            "continueOnFailure": false,
+            "isDisabled": false,
+            "parameters": {
+              "input": {
+                "content": "The unblacklist action failed for <i>{{[a62306f8-17fb-4ab7-8228-348bd1433df7].[hash]}}</i>\n<br>\nSomething went wrong while unblacklisting. Double-check that an IP address or SHA1 indicator was provided. If that's correct, please check the job details for more information.\n<br><br>\n{{[$job].[URL]}}"
```
Comment (mrinehart-r7) [inline workflows/Blacklist_Indicators_with_Trend_Micro_Apex_from_Microsoft_Teams/Blacklist_Indicators_with_Trend_Micro_Apex_from_Microsoft_Teams.icon:1512]:
```suggestion
                "content": "The unblacklist action failed for <i>{{[a62306f8-17fb-4ab7-8228-348bd1433df7].[hash]}}</i>\n<br>\nSomething went wrong while unblacklisting. Double-check that an IP address or SHA-1 hash indicator was provided. If that's correct, please check the job details for more information.\n<br><br>\n{{[$job].[URL]}}"
```

File: workflows/Blacklist_Indicators_with_Trend_Micro_Apex_from_Microsoft_Teams/Blacklist_Indicators_with_Trend_Micro_Apex_from_Microsoft_Teams.icon [hunk 4]
```diff
@@ -0,0 +1,2204 @@
...
+              "type": "object"
+            },
+            "outputJSONSchema": null,
+            "defaultImageData": "",
+            "connectionType": "CONNECTION"
+          },
+          "c6b577e3-9857-4c0e-9c30-f1efe042f5a5": {
+            "nodeId": "c6b577e3-9857-4c0e-9c30-f1efe042f5a5",
+            "name": "Blacklist Request",
+            "description": "Microsoft Teams message that triggers if a blacklist-indicator or unblacklist-indicator command is found containing a SHA1 hash or IP address",
```
Comment (mrinehart-r7) [inline workflows/Blacklist_Indicators_with_Trend_Micro_Apex_from_Microsoft_Teams/Blacklist_Indicators_with_Trend_Micro_Apex_from_Microsoft_Teams.icon:1771]:
```suggestion
            "description": "Microsoft Teams message that triggers if a blacklist-indicator or unblacklist-indicator command is found containing a SHA-1 hash or IP address",
```

File: workflows/Blacklist_Indicators_with_Trend_Micro_Apex_from_Microsoft_Teams/Blacklist_Indicators_with_Trend_Micro_Apex_from_Microsoft_Teams.icon [hunk 5]
```diff
@@ -0,0 +1,2204 @@
...
+          },
+          "d3e33086-f4e3-42f8-95ed-59f50c29c72e": {
+            "nodeId": "d3e33086-f4e3-42f8-95ed-59f50c29c72e",
+            "name": "Indicator Not Found",
+            "type": "artifact",
+            "continueOnFailure": false,
+            "isDisabled": false,
+            "parameters": {
+              "input": {
+                "content": "This workflow will blacklist and unblacklist indicators in Trend Micro Apex. To use this workflow provide a blacklist or unblacklist command in Microsoft Teams with a supported indicator of an IP address or SHA1 hash.\n\n<b>Invalid indicator identified, please provide a valid indicator.</b>\n<br><br>\nFor example:\n<br><br>\n<ul>\n<li>!blacklist-indicator a67f7c65c8ac1788a41be05f39d083ff532811a5</li>\n<li>!unblacklist-indicator a67f7c65c8ac1788a41be05f39d083ff532811a5</li>\n<li>!blacklist-indicator 198.51.100.10</li>\n<li>!unblacklist-indicator 198.51.100.10</li>\n</ul>"
```
Comment (mrinehart-r7) [inline workflows/Blacklist_Indicators_with_Trend_Micro_Apex_from_Microsoft_Teams/Blacklist_Indicators_with_Trend_Micro_Apex_from_Microsoft_Teams.icon:1947]:
```suggestion
                "content": "This workflow will blacklist and unblacklist indicators in Trend Micro Apex. To use this workflow provide a blacklist or unblacklist command in Microsoft Teams with a supported indicator of an IP address or SHA-1 hash.\n\n<b>Invalid indicator identified, please provide a valid indicator.</b>\n<br><br>\nFor example:\n<br><br>\n<ul>\n<li>!blacklist-indicator a67f7c65c8ac1788a41be05f39d083ff532811a5</li>\n<li>!unblacklist-indicator a67f7c65c8ac1788a41be05f39d083ff532811a5</li>\n<li>!blacklist-indicator 198.51.100.10</li>\n<li>!unblacklist-indicator 198.51.100.10</li>\n</ul>"
```

File: workflows/Blacklist_Indicators_with_Trend_Micro_Apex_from_Microsoft_Teams/Blacklist_Indicators_with_Trend_Micro_Apex_from_Microsoft_Teams.icon [hunk 6]
```diff
@@ -0,0 +1,2204 @@
...
+          },
+          "de9f616b-740a-4e2e-a506-33874749e04a": {
+            "nodeId": "de9f616b-740a-4e2e-a506-33874749e04a",
+            "name": "Blacklist Failure",
+            "type": "artifact",
+            "continueOnFailure": false,
+            "isDisabled": false,
+            "parameters": {
+              "input": {
+                "content": "The blacklist action failed for <i>{{[a62306f8-17fb-4ab7-8228-348bd1433df7].[indicator]}}</i>\n<br><br>\nSomething went wrong while blacklisting. Double-check that an IP address or SHA1 indicator was provided. If that's correct, please check the job details for more information.\n<br><br>\n{{[$job].[URL]}}"
```
Comment (mrinehart-r7) [inline workflows/Blacklist_Indicators_with_Trend_Micro_Apex_from_Microsoft_Teams/Blacklist_Indicators_with_Trend_Micro_Apex_from_Microsoft_Teams.icon:1983]:
```suggestion
                "content": "The blacklist action failed for <i>{{[a62306f8-17fb-4ab7-8228-348bd1433df7].[indicator]}}</i>\n<br><br>\nSomething went wrong while blacklisting. Double-check that an IP address or SHA-1 hash indicator was provided. If that's correct, please check the job details for more information.\n<br><br>\n{{[$job].[URL]}}"
```

File: workflows/Blacklist_Indicators_with_Trend_Micro_Apex_from_Microsoft_Teams/Blacklist_Indicators_with_Trend_Micro_Apex_from_Microsoft_Teams.icon [hunk 7]
```diff
@@ -0,0 +1,2204 @@
...
+        "humanCostSeconds": 0,
+        "humanCostDisplayUnit": ""
+      }
+    ],
+    "triggers": [
+      {
+        "id": "d2ceb88e-4220-45ba-8ba8-67b2f6e3727b",
+        "identifier": "new_message_received",
+        "name": "Blacklist Request",
+        "description": "Microsoft Teams message that triggers if a blacklist-indicator or unblacklist-indicator command is found containing a SHA1 hash or IP address",
```
Comment (mrinehart-r7) [inline workflows/Blacklist_Indicators_with_Trend_Micro_Apex_from_Microsoft_Teams/Blacklist_Indicators_with_Trend_Micro_Apex_from_Microsoft_Teams.icon:2042]:
```suggestion
        "description": "Microsoft Teams message that triggers if a blacklist-indicator or unblacklist-indicator command is found containing a SHA-1 hash or IP address",
```

File: workflows/Blacklist_Indicators_with_Trend_Micro_Apex_from_Microsoft_Teams/help.md [hunk 8]
```diff
@@ -0,0 +1,61 @@
+# Description
+
+This workflow blacklists or unblacklists indicators with Trend Micro Apex via Microsoft Teams command and reports information back.
+Indicators supported in this workflow are SHA1 hashes and IP addresses.
```
Comment (mrinehart-r7) [inline workflows/Blacklist_Indicators_with_Trend_Micro_Apex_from_Microsoft_Teams/help.md:4]:
```suggestion
Indicators supported in this workflow are SHA-1 hashes and IP addresses.
```

File: workflows/Blacklist_Indicators_with_Trend_Micro_Apex_from_Microsoft_Teams/workflow.spec.yaml [hunk 9]
```diff
@@ -0,0 +1,31 @@
+extension: workflow
+products: ["insightconnect"]
+name: Blacklist_Indicators_with_Trend_Micro_Apex_from_Microsoft_Teams
+title: Blacklist Indicators with Trend Micro Apex from Microsoft Teams
+description: Use the simplicity of Microsoft Teams to blacklist and unblacklist indicators such as SHA1 hashes and IP addresses with Trend Micro Apex.
```
Comment (mrinehart-r7) [inline workflows/Blacklist_Indicators_with_Trend_Micro_Apex_from_Microsoft_Teams/workflow.spec.yaml:5]:
```suggestion
description: Use the simplicity of Microsoft Teams to blacklist and unblacklist indicators such as SHA-1 hashes and IP addresses with Trend Micro Apex.
```

**Comment by jschipp-r7** [comment]
Applied all suggestions

**Pull request merged**